### PR TITLE
Issue #618 Update "writing and publishing docs" topic

### PR DESCRIPTION
**Pull request by thatdocslady**
Fixes issue #618 

This PR supersedes previous PR #836 and includes all the feedback cycles from the team and the latest patch from @hferentschik.

### CODE REVIEW

File: docs/source/developing/writing-docs.adoc [hunk 1]
```diff
@@ -66,39 +113,48 @@ To build and serve the documentation for editing, run:
...
 
 ----
-$ make gen_adoc_tar
+$ make clean_docs
 ----
 
-Once the build completes there will be a _minishift-adoc.tar_ in the _docs/build_ directory
-of your checkout.
+[[auto-deploy-to-docs-openshift-org]]
+== Automatic deployment to docs.openshift.org
```
Comment (hferentschik) [inline docs/source/developing/writing-docs.adoc]: Can we be a bit more neutral here. Deployment is not the key here, nor is it automatic. I suggest:

"Integration with docs.openshift.org"

File: docs/source/developing/writing-docs.adoc [hunk 2]
```diff
@@ -10,29 +10,73 @@ toc::[]
...
+[[building-docs]]
+== Building the documentation
+
+By default, the documentation is built in a Docker container. This way
+you do not need to install all the required dependencies on your
+development machine. All you need is a running Docker daemon. In case
+you don't have one, use Minishift itself. For more information, see
+xref:../using/docker-daemon.adoc#reusing-docker-daemon[reusing the Docker daemon].
+
+To generate the documentation into the directory _docs/build_, run:
```
Comment (hferentschik) [inline docs/source/developing/writing-docs.adoc]: *_docs/build_* :-)

**Comment by thatdocslady** [comment]
@hferentschik implemented your feedback from the comments and also from IRC (changed back to "integration" because it does make sense).

**Comment by hferentschik** [comment]
>  implemented your feedback

thanks

**Comment by hferentschik** [comment]
Merged, thanks.

**Comment by hferentschik** [comment]
Thanks for hanging in on this one.

**Pull request merged**